src/main/java/com/google/sps/data/Election.java

```diff
@@ -134,92 +130,43 @@ public static Election fromElectionQuery(JSONObject electionQueryData) throws JS
    * @param datastore the Datastore to store this election's list of contests
    * @param voterInfoQueryData the JSON output of a voterInfoQuery to the Google Civic Information
    *     API
+   * @param divisions the set of divisions which limit the contests to be added
    * @return the new Election object
    */
-  public Election fromVoterInfoQuery(DatastoreService datastore, JSONObject voterInfoQueryData)
+  public Election fromVoterInfoQuery(
+      DatastoreService datastore, JSONObject voterInfoQueryData, Set<String> divisions)
       throws JSONException {
     Set<Long> contestKeyList = this.getContests();
     Set<Long> referendumKeyList = this.getReferendums();
-
-    List<EmbeddedEntity> pollingStations = new ArrayList<>(this.getPollingStations());
-    ImmutableSet<EmbeddedEntity> pollingStationSet;
+    Set<String> divisionsList = this.getDivisions();
+    divisionsList.addAll(divisions);
 
     if (voterInfoQueryData.has(CONTESTS_JSON_KEYWORD)) {
```

Comment:
This nesting has gotten pretty deep. That's bad, see:
https://drive.google.com/file/d/0B-KFDiTrQe9lLU9uVDZzZWNmRVU/view?usp=sharing

I think we can refactor this like:
------------------

if (!voterInfoQueryData.has(CONTESTS_JSON_KEYWORD)) {
return this; // nothing will get added, so just return early 
}

// ...
for (Object contestObject : contestListData) {
  if (divisions.contains(currentDivision)) {
    continue;
  }

  if (contest.getString(Contest.TYPE_JSON_KEYWORD).equals(Referendum.ENTITY_KIND)) {
    //...
  } else {
    //...
  }
}

Reply:
done